sympy/solvers/solvers.py

```diff
@@ -1557,7 +1557,11 @@ def _expand(p):
                         try:
                             t = Dummy('t')
                             iv = _solve(u - t, symbol, **flags)
-                            soln = list(ordered({i.subs(t, s) for i in iv for s in soln}))
+                            new_soln = list(ordered({i.subs(t, s) for i in iv for s in soln}))
+                            if new_soln[0] == S.NaN:
```

Comment:
I'm unsure of this. Firstly, it works in this case because `iv` contains division by `log(_t)` and there is only one element in `new_soln`. But suppose `new_soln` contains several elements, and some of them become "S.NaN" after substitution, but not `new_soln[0]`. I suppose in that case we'll need to remove `S.NaN`s from `new_soln` but what to do with the rest? We could return them but, like in this case, the rest of `_solve()` could find more solutions later on. Maybe keep them and let `solve()` look further anyway?
Secondly, does a similar check need to be added for the case `len(gens) > 1`?
Also, there is the performance issue. Don't know if it's important, but letting line 1559 run can unnecessarily increase the time if `u-t` is just the general case of the original equation. I've checked, and for this case what you have takes about `0.45` seconds, and about `0.31` if you don't let line 1559 run (by checking first that `u` is linear).